src/Compilers/Core/Portable/MetadataReference/AssemblyMetadata.cs

```diff
@@ -260,11 +260,6 @@ internal new AssemblyMetadata Copy()
             return new AssemblyMetadata(this, shareCachedSymbols: true);
         }
 
-        internal AssemblyMetadata CopyWithoutSharingCachedSymbols()
```

Comment:
>internal AssemblyMetadata CopyWithoutSharingCachedSymbols() [](http://example.com/codeflow?start=8&length=59)

Given the volume of changes in this PR, consider splitting it into several more targeted PRs. We can discuss offline in more details.